pkg/app/AppService.go

```diff
@@ -2267,15 +2448,45 @@ func (impl *AppServiceImpl) getConfigMapAndSecretJsonV2(appId int, envId int, pi
 	if err != nil {
 		return []byte("{}"), err
 	}
+
+	if configMapJson != "" && len(entityToVariablesCM) > 0 {
+		resolvedTemplateCM, variableMapCM, err = impl.GetResolvedTemplateAndVariableMap(scope, string(configMapByte), entitiesForCM, entityToVariablesCM)
+		cMCSJson.envOverride.ResolvedEnvOverrideValuesForCM = resolvedTemplateCM
+		cMCSJson.envOverride.VariableSnapshotForCM = variableMapCM
+	}
+	var encodedData string
+	if err != nil {
+		return nil, err
+	}
 	secretDataByte, err := json.Marshal(secretResponseR)
 	if err != nil {
-		return []byte("{}"), err
+		return secretDataByte, err
+
+	}
+	if secretDataJson != "" && len(entityToVariablesCS) > 0 {
+		data, err := bean.GetDecodedDataForSecret(string(secretDataByte))
+		if err != nil {
+			return nil, err
+		}
+		resolvedTemplateCS, variableMapCS, err = impl.GetResolvedTemplateAndVariableMap(scope, data, entitiesForCS, entityToVariablesCS)
+		cMCSJson.envOverride.ResolvedEnvOverrideValuesForCS = resolvedTemplateCS
+		cMCSJson.envOverride.VariableSnapshotForCS = variableMapCS
+		encodedData, err = bean.GetEncodedDataForSecret(resolvedTemplateCS)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	if err != nil {
+		return nil, err
 	}
 
-	merged, err = impl.mergeUtil.JsonPatch(configMapByte, secretDataByte)
+	merged, err = impl.mergeUtil.JsonPatch([]byte(cMCSJson.envOverride.ResolvedEnvOverrideValuesForCM), []byte(encodedData))
```

Comment:
how is unresolved data getting saved into history table?

Reply:
discussed